src/app/api/webhook/webhook.service.ts

```diff
@@ -121,16 +121,31 @@ class WebhookService extends BaseService {
   }
 
   async handleUserDeleted(assigneeId: string, assigneeType: AssigneeType) {
-    if (assigneeType === AssigneeType.company) return
-
     const tasksService = new TasksService(this.user)
     // Delete corresponding tasks
     console.info(`Deleting all tasks for ${assigneeType} ${assigneeId}`)
-    await tasksService.deleteAllAssigneeTasks(assigneeId, assigneeType)
+    const tasks = await tasksService.deleteAllAssigneeTasks(assigneeId, assigneeType)
 
-    // So how do we delete corresponding notifications, you ask? Good question.
-    // Copilot calls multiple `client.updated` when unassigning clients companies after `company.deleted` is called.
-    // So handling deleting company task notifications should be done in `client.updated` company unassignment logic instead
+    // Now delete any and all associated notifications triggered on behalf of company tasks for clients.
+    // i.e. decrement client task count in CU portal
+    if (assigneeType === AssigneeType.company) {
+      const deletedTaskIds = tasks.map((task) => task.id)
+      await this.handleDeletingTaskNotifications(deletedTaskIds)
+    }
+  }
+
+  private async handleDeletingTaskNotifications(taskIds: string[]) {
+    const clientTaskNotifications = await this.db.clientNotification.findMany({
+      where: {
+        taskId: { in: taskIds },
+      },
+    })
+
+    const clientNotificationIds = clientTaskNotifications.map((notification) => notification.notificationId)
+    if (clientNotificationIds.length) {
+      await this.copilot.bulkMarkNotificationsAsRead(clientNotificationIds)
```

Comment:
can we run these operations in async?

```
Promise.all([
    this.copilot.bulkMarkNotificationsAsRead(clientNotificationIds),
    this.db.clientNotification.deleteMany({ where: { notificationId: { in: clientNotificationIds } } })
])
```

Reply:
> or do we need to delete only after marking them as read?

Yes, not doing this could cause inconsistencies, and I don't think doing them in parallel for each notification (markAsRead -> delete from db) would be very performant, although that's the best way to approach this.

Same for not doing those async operations in parallel. Doing markAsRead again to a read notification won't cause issues, but deleting them before they are marked as read in a different thread definitely will.

With our current promise-retry mechanism this should be very reliable.